### PR TITLE
fix: not update overdue events on search screen

### PR DESCRIPTION
**Pull request by mmmateos**
Jira Issue: https://jira.dhis2.org/browse/ANDROAPP-2604

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java [hunk 1]
```diff
@@ -617,18 +609,10 @@ else if (d2.trackedEntityModule().trackedEntityAttributes().byUid().eq(tetAttrib
...
-                    } catch (D2Error d2Error) {
-                        d2Error.printStackTrace();
-                    }
-                }
-
                 boolean hasEventWithStatus = !d2.eventModule().events().byTrackedEntityInstanceUids(Collections.singletonList(tei.uid())).byStatus().in(eventStatuses).blockingIsEmpty();
-                if (!hasEventWithStatus)
+                boolean hasScheduledEvent = !d2.eventModule().events().byTrackedEntityInstanceUids(Collections.singletonList(tei.uid())).byStatus().eq(EventStatus.SCHEDULE)
+                        .byDueDate().before(new Date()).blockingIsEmpty();
+                if (!hasEventWithStatus && !hasScheduledEvent)
```
Comment (MarinaCape) [inline app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java]: The "if" should has brackets

**Comment by Balcan** [comment]
Tested

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java (hunk 1 above)
Comment (JaimeToca) [reply app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java]: Summon @ferdyrod  :imp:

Comment (JaimeToca) [reply app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java]: xD

**Pull request merged**